controller/backup_controller.go

```diff
@@ -317,6 +319,11 @@ func (bc *BackupController) reconcile(backupName string) (err error) {
 		return bc.ds.RemoveFinalizerForBackup(backup)
 	}
 
+	// Don't handle backups in final states
+	if !bc.backupNotInFinalState(backup) {
+		return nil
+	}
+
```

Comment:
This part needs to be put in [L355](https://github.com/longhorn/longhorn-manager/pull/2182/files#diff-eaf2136df9567f9934b678766cb7390741b265de41c1e5abb9ee2785dbe624e2L355) as it was

This is to fix the issue: BackupTarget becomes unavailable when Backup is being created
The issue happened because
1. Backup creation is an async call, so it will keep uploading in Engine/Replica
2. Backup creation status will keep in the Engine memory which will sync with Replica
3. Backup Controller will sync with the status in Engine memory. So even BackupTarget is unavailable, it will keep uploading the Backup and the updating the progress in CRD [here](https://github.com/longhorn/longhorn-manager/blob/master/controller/backup_controller.go#L386)

The problem is when Backup is Completed, we do BackupInspect() which will fail because Target is now unavailable.
So the backup.Status.LastSyncedAt is never set, we only set backup.Status.State=Completed.

If we put this check here. Then even when the BackupTarget is available again, it still won't reach the following logic to inspect the info from the BackupStore and won't get `backup.Status.LastSyncedAt` assigned.

Its not "Don't handle backup in final states". (Note: Its okay to do that, because we only inspect the info from backupstore and sync to the CR status.)
But "**Don't retry creating the Backup if backup is in final states already**"

Reply:
Moved it back. Thanks for your explanation.